spec/features/posts_index_spec.rb

```diff
@@ -0,0 +1,60 @@
+require 'rails_helper'
+
+describe 'Users Post Index', type: :feature do
+  before :each do
+    @user = User.create(name: 'Muhammed-ali', photo: 'https://Akaiiii.jpg', bio: 'Martial Artist')
+    @user2 = User.create(name: 'Bonzai73', photo: 'https://bon-bon.jpg', bio: 'Monk of the mountains')
+    @post1 = Post.create(author_id: @user.id, title: 'The way of the Water',
+                         text: 'In martial arts, the way of the water is the way of the soft and yielding')
+    @post2 = Post.create(author_id: @user.id, title: 'Amaterasu',
+                         text: 'Amaterasu is the goddess of the sun and the universe, the ultimate source of all life')
+    @post3 = Post.create(author_id: @user.id, title: 'Legacy', text: 'The legacy of the righteous is a blessing')
+    Comment.create(user_id: @user2.id, post_id: @post1.id, text: 'I like this post')
+    Comment.create(user_id: @user.id, post_id: @post1.id, text: 'Thanks! I wrote it myself based on my experiences')
+    Like.create(user_id: @user2.id, post_id: @post1.id)
+    Like.create(user_id: @user.id, post_id: @post1.id)
+    visit user_posts_path(@user.id)
+  end
+
+  scenario "I can see the user's profile picture" do
+    expect(page).to have_css("img[src*='https://Akaiiii.jpg']")
+  end
+
+  scenario "I can see the user's username" do
+    expect(page).to have_content('Muhammed-ali')
+  end
+
+  scenario 'I can see the number of posts the user has written' do
+    expect(page).to have_content('Number of posts: 3')
+  end
+
+  scenario "I can see a post's title" do
+    expect(page).to have_content('The way of the Water')
+  end
+
+  scenario "I can see some of the post's body" do
+    expect(page).to have_content('In martial arts, the way of the water is the way of the soft and yielding')
+  end
+
+  scenario 'I can see the first comments on a post' do
+    expect(page).to have_content('I like this post')
+  end
+
+  scenario 'I can see how many comments a post has' do
+    expect(page).to have_content('Comments: 2')
+  end
+
+  scenario 'I can see how many likes a post has' do
+    expect(page).to have_content('Likes: 2')
+  end
+
+  scenario "When I click on a post, it redirects me to that post's show page" do
+    click_on('The way of the Water')
+    expect(page).to have_content('In martial arts, the way of the water is the way of the soft and yielding')
+  end
+
+  scenario 'I can see the pagination' do
+    click_on('Back')
+    expect(page).to have_content('See all posts')
+  end
```

Comment:
- [ ] As requested by all the previous CR please 🙏🏽 add a test to verify that there is a pagination button that allows showing the rest of the post when they exed the view, first of all, we need to create the Pagination button as shown in the wireframe, e.g I have created a user with 227 post and for a better UX I need a pagination button that allows the user check the post ten by ten and avoid scrolling down too long time until the end 👇🏽 

![image](https://github.com/matee-safi/Blog-app/assets/58642949/f2c9ff85-a7e0-492f-b2b5-b785ec390fd0)

![image](https://github.com/matee-safi/Blog-app/assets/58642949/084218a8-9fcf-422b-9259-d10eabc4f367)

![image](https://github.com/matee-safi/Blog-app/assets/58642949/18dfdf7d-af22-46f2-8963-30cd4f450c61)